autoscheduler/frontend/src/tests/ReduxAvailability.test.tsx

```diff
@@ -0,0 +1,164 @@
+import { createStore } from 'redux';
+import autoSchedulerReducer from '../redux/reducers';
+import {
+  addAvailability,
+  deleteAvailability,
+  updateAvailability,
+} from '../redux/actions';
+import 'isomorphic-fetch';
+import Availability, { AvailabilityType, argsToAvailability } from '../types/Availability';
+import DayOfWeek from '../types/DayOfWeek';
+
+/**
+ * Converts a pair of hours and minutes into a number of minutes past midnight
+ */
+const makeTime = (h: number, m: number): number => h * 60 + m;
+/**
+ * Makes availabilities "Busy on Wednesday", since that part isn't being tested
+ */
+const dummyArgs = {
+  available: AvailabilityType.BUSY,
+  dayOfWeek: DayOfWeek.WED,
+};
+describe('Availabilities', () => {
+  describe('of the same type are merged', () => {
+    test('with an overlap on one end', () => {
+      // arrange
+      const store = createStore(autoSchedulerReducer);
+      const availability1 = {
+        ...dummyArgs,
+        time1: makeTime(8, 0),
+        time2: makeTime(12, 0),
+      };
+      const availability2 = {
+        ...dummyArgs,
+        time1: makeTime(11, 30),
+        time2: makeTime(13, 42),
+      };
+
+      // act
+      store.dispatch(addAvailability(availability1));
+      store.dispatch(addAvailability(availability2));
+
+      // assert
+      expect(store.getState().availability).toEqual([{
+        available: AvailabilityType.BUSY,
+        dayOfWeek: DayOfWeek.WED,
+        startTimeHours: 8,
+        startTimeMinutes: 0,
+        endTimeHours: 13,
+        endTimeMinutes: 42,
+      }]);
+    });
+    test('with overlaps on both ends', () => {
+      // arrange
+      const store = createStore(autoSchedulerReducer);
+      const availability1 = {
+        ...dummyArgs,
+        time1: makeTime(11, 30),
+        time2: makeTime(13, 42),
+      };
+      const availability2 = {
+        ...dummyArgs,
+        time1: makeTime(7, 0),
+        time2: makeTime(8, 30),
+      };
+      const availability3 = {
+        available: AvailabilityType.BUSY,
+        dayOfWeek: DayOfWeek.WED,
+        time1: makeTime(8, 0),
+        time2: makeTime(12, 0),
+      };
+
+      // act
+      store.dispatch(addAvailability(availability1));
+      store.dispatch(addAvailability(availability2));
+      store.dispatch(addAvailability(availability3));
+
+      // assert
+      expect(store.getState().availability).toEqual([{
+        available: AvailabilityType.BUSY,
+        dayOfWeek: DayOfWeek.WED,
+        startTimeHours: 7,
+        startTimeMinutes: 0,
+        endTimeHours: 13,
+        endTimeMinutes: 42,
```

Comment:
This is a trivial change for me to make, but for improved understanding of good test formatting, could you explain why you arrange the tests this way? Is the `assert` step supposed to be pretty minimal?

Reply:
Also, would you prefer to have them called `expected` or something more descriptive like `mergedAvailability`?

Reply:
Yeah so I generally just like to have a consistent structure with the tests I write, so in the case of a test failing we can quickly open the test and see what the expected value is should be, rather than looking at the `assert` and seeing the variable we're comparing to (i.e. to manually find what the expected is). Basically, it's to save time, and in the case of naming it `mergedAvailability` I think it's probably just as descriptive as naming it `expected`, given that we're testing availability merging.

This is a really trivial / small thing and honestly I don't think it matters *that* much, and at the end of the day it's just personal preference. I just think having all of our tests that are doing some computation and comparing the result to the expected value name the expected value be `expected` gives a good consistency to our tests.

Reply:
Aight done